### PR TITLE
check for overlapping bookings

### PR DESCRIPTION
**Pull request by jamwalla**

### CODE REVIEW

File: recidiviz/persistence/persistence.py [hunk 1]
```diff
@@ -200,17 +200,23 @@ def _should_abort(
 
 # Note: If we ever want to validate more than the existence of multiple open
 # bookings, we should make validation an entirely separate module/step.
```
Comment (terinpw) [inline recidiviz/persistence/persistence.py]: lol should we move this into it's own file given the comments above...?

File: recidiviz/persistence/persistence.py [hunk 2]
```diff
@@ -200,17 +200,23 @@ def _should_abort(
...
     validated_people = []
     for person in people:
-        open_bookings = [booking for booking in person.bookings if
-                         persistence_utils.is_booking_active(booking)]
-        if len(open_bookings) > 1:
+        if persistence_utils.has_malformed_dates(person):
+            logging.error("Booking found with release date before admission "
+                          "date for person: %s", person.person_id)
+            data_validation_errors += 1
+        if persistence_utils.has_multiple_active_bookings(person):
```
Comment (terinpw) [inline recidiviz/persistence/persistence.py]: should we combine has_multiple_active_bookings into has_overlapping_bookings?

File: recidiviz/persistence/persistence.py (hunk 1 above)
Comment (jamwalla) [reply recidiviz/persistence/persistence.py]: 🙄

Comment (jamwalla) [reply recidiviz/persistence/persistence.py]: yes ofc!!! good call

File: recidiviz/persistence/persistence.py (hunk 2 above)
Comment (jamwalla) [reply recidiviz/persistence/persistence.py]: done by defaulting date ranges to datetime.date.min/max

File: recidiviz/tests/persistence/entity_validator_test.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import datetime
+import unittest
+
+from recidiviz.persistence import entities
+from recidiviz.persistence.entity_validator import _has_overlapping_bookings
+
+
+class EntityValidatorTest(unittest.TestCase):
+    """Tests entity validation."""
+    def test_has_overlapping_bookings(self):
```
Comment (terinpw) [inline recidiviz/tests/persistence/entity_validator_test.py]: I think we should add a test for malformed dates as well.

**Pull request merged**